presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ParametricScalarImplementation.java

```diff
@@ -320,10 +451,21 @@ private void parseArguments(Method method)
                             .findFirst()
                             .orElseThrow(() -> new IllegalArgumentException(format("Method [%s] is missing @SqlType annotation for parameter", method)));
                     TypeSignature typeSignature = parseTypeSignature(type.value(), literalParameters);
-                    boolean nullableArgument = Stream.of(annotations).anyMatch(SqlNullable.class::isInstance);
-                    checkArgument(nullableArgument || !containsLegacyNullable(annotations), "Method [%s] has parameter annotated with @Nullable but not @SqlNullable", method);
+                    if (Stream.of(annotations).anyMatch(SqlNullable.class::isInstance)) {
```

Comment:
Use a local variable for `nullConventionFlag` here  (instead of using the field)